pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -437,6 +438,16 @@ func updatePodFromAllocation(pod *v1.Pod, allocated state.PodResourceInfo) (*v1.
 	}
 
 	updated := false
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodLevelResourcesVerticalScaling) && utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources) {
+		pAlloc := allocated.PodLevelResources
+		if !apiequality.Semantic.DeepEqual(pod.Spec.Resources, pAlloc) {
+			if !updated {
```

Comment:
`updated` is always false at the line, isn't it? I don't think we need this `if`